lib/scholar/metrics.ex

```diff
@@ -153,9 +282,36 @@ defmodule Scholar.Metrics do
     |> Nx.sum()
   end
 
+  @doc """
+  Computes the number of true positive predictions with respect
+  to given targets for multi-class classification problems.
+
+  ## Options
+
+    * `:num_classes` - Number of classes contained in the input tensors
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([0, 1, 1, 1, 1, 0, 2, 1, 0, 1], type: {:u, 32})
+      iex> y_pred = Nx.tensor([0, 2, 1, 1, 2, 2, 2, 0, 0, 1], type: {:u, 32})
+      iex> Scholar.Metrics.true_positives(y_true, y_pred, num_classes: 3)
+      #Nx.Tensor<
+        s64[3]
+        [2, 3, 1]
+      >
+  """
+  defn true_positives(y_true, y_pred, opts \\ []) do
```

Comment:
Honestly, I think exposing tp, tn, fp, fn is too granular given that we have `confusion_matrix`.

Reply:
Yup, they also seem to be somewhat redundant/duplicated in terms of some calculations, and the calculated confusion matrix cannot easily be reused between them. It's simpler to derive tp, tn, fp, fn from the confusion matrix.